File: app/src/main/java/com/bale/deeplinkdemo/deeplinks/LinkDispatcherActivity.java
package com.bale.deeplinkdemo.deeplinks;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;

import com.bale.deeplinkdemo.BuildConfig;

public class LinkDispatcherActivity extends Activity {
    private final UriToIntentMapper mMapper = new UriToIntentMapper(this, new IntentHelper());

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        try {
            mMapper.dispatchIntent(getIntent());

        } catch (IllegalArgumentException iae) {
            // Malformed URL
            if (BuildConfig.DEBUG) {
                Log.e("Deep links", "Invalid URI", iae);
            }
        } finally {
            // Always finish the Activity so that it doesn't stay in our history
            finish();
        }
    }
}

File: app/src/main/java/com/bale/deeplinkdemo/deeplinks/LinkDispatcherActivity.kt
package com.bale.deeplinkdemo.deeplinks

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.widget.Toast
import com.bale.deeplinkdemo.BuildConfig
import com.bale.deeplinkdemo.DashboardActivity

class LinkDispatcherActivity : AppCompatActivity() {

    private val mMapper = UriToIntentMapper(this)
    private val deepLinkJourneyRequestCode = 1000

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        try {
            mMapper.dispatchIntent(intent)
        } catch (iae: IllegalArgumentException) {
            if (BuildConfig.DEBUG) {
                Log.e("Deep links", "Invalid URI", iae)
            }
        } finally {
            // Always finish the Activity so that it doesn't stay in our history
            finish()
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == deepLinkJourneyRequestCode) {
            Toast.makeText(this, "Journey Finished", Toast.LENGTH_LONG).show()
            val intent = Intent(this, DashboardActivity::class.java)
            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
            startActivity(intent)
            finish()
        }
    }
}